main.go

```diff
@@ -38,45 +39,79 @@ var (
 	signingNameOverride = kingpin.Flag("name", "AWS Service to sign for").String();
 	hostOverride = kingpin.Flag("host", "Host to proxy to").String();
 	regionOverride = kingpin.Flag("region", "AWS region to sign for").String();
+	configSets = kingpin.Flag("config-set", "Host-based configuration overrides for role-arn/name/host/region (encoded as yaml)").Short('c').Strings()
 )
 
-func main() {
-	kingpin.Parse()
-
-	log.SetLevel(log.InfoLevel)
-	if *debug {
-		log.SetLevel(log.DebugLevel)
-	}
 
+func getSigner(roleArn string) *v4.Signer {
     session, err := session.NewSession()
     if err != nil {
         log.Fatal(err)
     }
 
 	var credentials *credentials.Credentials
-	if *roleArn != "" {
-	    credentials = stscreds.NewCredentials(session, *roleArn, func(p *stscreds.AssumeRoleProvider) {
+	if roleArn != "" {
+	    credentials = stscreds.NewCredentials(session, roleArn, func(p *stscreds.AssumeRoleProvider) {
 		    p.RoleSessionName = roleSessionName()
 	    })
 	} else {
 	    credentials = session.Config.Credentials
 	}
 
-	signer := v4.NewSigner(credentials)
+	return v4.NewSigner(credentials)
+}
+
+func main() {
+	kingpin.Parse()
+
+	log.SetLevel(log.InfoLevel)
+	if *debug {
+		log.SetLevel(log.DebugLevel)
+	}
+
+	signer := getSigner(*roleArn)
 
 	log.WithFields(log.Fields{"StripHeaders": *strip}).Infof("Stripping headers %s", *strip)
 	log.WithFields(log.Fields{"port": *port}).Infof("Listening on %s", *port)
 
+	clients := map[string]handler.Client{
+		"default": &handler.ProxyClient{
+			Signer: signer,
+			Client: http.DefaultClient,
+			StripRequestHeaders: *strip,
+			SigningNameOverride: *signingNameOverride,
+			HostOverride: *hostOverride,
+			RegionOverride: *regionOverride,
+		},
+	}
+	for _, configSetYaml := range *configSets {
+		configSet := handler.ConfigSet{}
+
+		err := yaml.Unmarshal([]byte(configSetYaml), &configSet)
+		if err != nil {
+			log.Fatalf("error parsing config set: %v", err)
+		}
+
+		log.WithFields(log.Fields{
+			"Host": configSet.Host,
+			"Name": configSet.Name,
+			"Region": configSet.Region,
+			"RoleArn": configSet.RoleArn,
+		}).Infof("Adding config for host")
+
+		clients[configSet.Host] = &handler.ProxyClient{
+			Signer: getSigner(configSet.RoleArn),
+			Client: http.DefaultClient,
+			StripRequestHeaders: *strip,
+			SigningNameOverride: configSet.Name,
+			HostOverride: configSet.Host,
+			RegionOverride: configSet.Region,
+		}
+	}
+
```

Comment:
Run gofmt on the code please

Reply:
The problem is that the existing code is not formatted using `gofmt`, so running it changes basically all of the existing code